external/anomaly/tests/test_ote_training.py

```diff
@@ -0,0 +1,385 @@
+                                                       OTETestTrainingAction,
+                                                       BaseOTETestAction,
+                                                       OTETestTrainingEvaluationAction,
+                                                       OTETestExportAction,
+                                                       OTETestExportEvaluationAction,
+                                                       OTETestPotAction,
+                                                       OTETestPotEvaluationAction,
+                                                       OTETestNNCFAction,
+                                                       OTETestNNCFEvaluationAction,                                                       
+                                                       OTETestNNCFExportAction,
+                                                       OTETestNNCFExportEvaluationAction)
+
+logger = get_logger(__name__)
+
+def DATASET_PARAMETERS_FIELDS() -> List[str]:
+    return deepcopy(['dataset_path',
+                     ])
+
+DatasetParameters = namedtuple('DatasetParameters', DATASET_PARAMETERS_FIELDS())
+
+
+def _get_dataset_params_from_dataset_definitions(dataset_definitions, dataset_name):
+    if dataset_name not in dataset_definitions:
+        raise ValueError(f'dataset {dataset_name} is absent in dataset_definitions, '
+                         f'dataset_definitions.keys={list(dataset_definitions.keys())}')
+    cur_dataset_definition = dataset_definitions[dataset_name]
+    training_parameters_fields = {k: v for k, v in cur_dataset_definition.items()
+                                  if k in DATASET_PARAMETERS_FIELDS()}
+    print(f"training_parameters_fields: {training_parameters_fields}")
+    make_paths_be_abs(training_parameters_fields, dataset_definitions[ROOT_PATH_KEY])
+    print(f"training_parameters_fields after make_paths_be_abs: {training_parameters_fields}")
+
+    assert set(DATASET_PARAMETERS_FIELDS()) == set(training_parameters_fields.keys()), \
+            f'ERROR: dataset definitions for name={dataset_name} does not contain all required fields'
+    assert all(training_parameters_fields.values()), \
+            f'ERROR: dataset definitions for name={dataset_name} contains empty values for some required fields'
+
+    params = DatasetParameters(**training_parameters_fields)
+    return params
+
+
+def _create_anomaly_classification_dataset_and_labels_schema(dataset_params, dataset_name):
+    logger.debug(f'Path to dataset: {dataset_params.dataset_path}')
+    category_list = [f.path for f in os.scandir(dataset_params.dataset_path) if f.is_dir()]
+    items = []
+    if "short" in dataset_name:
+        logger.debug(f'Creating short dataset {dataset_name}')
+        items.extend(OteMvtecDataset(path=dataset_params.dataset_path, seed=0).generate())
+    else:
+        for category in category_list:
+            if "vivo" not in category:
```

Comment:
Why such special branch for this case?

Reply:
'vivo' folder contains  .jpg files, not .png as other. So, to save dataset in initial condition, I decided to add this condition.